src/main/kotlin/com/pubnub/api/PubNub.kt

```diff
@@ -161,6 +170,7 @@ class PubNub internal constructor(
     internal val tokenManager: TokenManager = TokenManager()
     private val tokenParser: TokenParser = TokenParser()
     internal val listenerManager = ListenerManager(this)
+    private val eventEmitter = EventEmitterImpl(AnnouncementCallback.Phase.SUBSCRIPTION, ::accepts) // todo I tried to used it but doesn't work yet
```

Comment:
maybe you forgot to `listenerManager.addAnnouncementCallback(eventEmitter)` ?

Reply:
Thanks. That was it.